File: src/demo06_get_pose/src/get_pose.cpp
#include "ros/ros.h"
#include "turtlesim/Pose.h"

void doPose(const turtlesim::Pose::ConstPtr &p)
{
    ROS_INFO("x = %.2f y = %.2f theta = %.2f linear_velocity = %.2f angular_velocity = %.2f",
    p -> x, p -> y, p -> theta, p -> linear_velocity, p -> angular_velocity);
}
int main(int argc, char *argv[])
{
    ros::init(argc, argv, "get_pose");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe<turtlesim::Pose>("/turtle1/pose", 1000, doPose);
    ros::spin();
    return 0;
}

File: src/demo06_get_pose/scripts/get_pose.py
#! /usr/bin/env python
import rospy
from turtlesim.msg import Pose

def doPose(data):
    rospy.loginfo(f"x = {data.x}, y = {data.y},  = {data.theta}, linear_velocity = {data.linear_velocity}, angular_velocity = {data.angular_velocity}");

if __name__ == "__main__":
    rospy.init_node("get_pose")
    sub = rospy.Subscriber("/turtle1/pose", Pose, doPose, queue_size=1000)
    rospy.spin()